src/include/cell_inline.h

```diff
@@ -189,13 +189,41 @@ __cell_pack_addr_validity(WT_SESSION_IMPL *session, uint8_t **pp, WT_TIME_AGGREG
     *flagsp = flags;
 }
 
+#ifdef WT_STANDALONE_BUILD_FT_FIX
+/*
+ * __cell_pack_addr_del --
+ *     Pack the fast-delete information for an address.
+ */
+static inline void
+__cell_pack_addr_del(uint8_t **pp, WT_PAGE_DELETED *page_del, u_int cell_type)
+{
+    uint8_t v;
+
+    WT_IGNORE_RET(__wt_vpack_uint(pp, 0, page_del->txnid));
+    WT_IGNORE_RET(__wt_vpack_uint(pp, 0, page_del->timestamp));
+    WT_IGNORE_RET(__wt_vpack_uint(pp, 0, page_del->durable_timestamp));
+
+    /*
+     * There's two additional bits of information: is the fast-delete prepared, and are we replacing
+     * a leaf without overflow items or not. (There are 6 additional bits here for later.)
+     */
+    v = 0;
+    if (cell_type == WT_CELL_ADDR_LEAF_NO)
+        FLD_SET(v, WT_CELL_ADDR_DEL_LEAF_NO);
+    if (page_del->prepare_state != 0)
```

Comment:
There is no macro defined for unprepared state? What about the case the prepared is resolved? In the case we should not treat it differently right?

Reply:
That's a good catch -- I agree it's likely wrong as is, but I'm not sure what it should be.

If a checkpoint is writing this information, then `prepare_state` could be any of "inprogress", "locked", or "resolved". 

I'm going to need some help, I don't know what is the correct thing to write here. The test to write a proxy child is `__wt_page_del_active(session, ref, true)`, which is always going to return `true` if the prepare isn't resolved.

Reply:
@quchenhao, I just pushed a change in 6eb1285 that makes the situation better. It asserts the prepare is in-progress as a placeholder, even though that's not correct.

I also cleaned up some field usage that was from a previous version -- I don't think we need to know if this is a leaf, or a leaf without overflow items, but I will stare at it a little more tomorrow.

Reply:
I don't understand why we are not just checking prepare equals locked or prepared state? Is this because it can be committed concurrently to the check?

Reply:
We're checkpointing the internal page and we have the child WT_REF locked, and we're looking at the child WT_REF state. I don't know if a prepare can be updated without holding the WT_REF locked or not -- I need some help here. I'm happy to talk through what's happening, but I don't understand how prepare is handled sufficiently to make these changes on my own.